lib/db/mysql.js

```diff
@@ -413,92 +407,148 @@ module.exports = function (log, error) {
     })
   }
 
-  // This function takes a result and unwraps comma-separated values to Arrays
-  // and eventually applies transforms.
-  function unwrapArrays(result, fields, separator = ',') {
-    for (const {name, transform} of fields) {
-      const arr = result[name] ? result[name].split(separator) : []
-      result[name] = transform ? transform(arr) : arr
-    }
-  }
-
-  function unwrapCapabilities(result, fieldName) {
-    const field = {
-      name: fieldName,
-      transform(capabilityIds) {
-        return capabilityIds.reduce((acc, c) => {
-          const capabilityStr = dbUtil.mapDeviceCapability(parseInt(c))
-          if (capabilityStr != null) {
-            acc.push(capabilityStr)
+  // A helper function for aggregating name:value pairs into a JSON object.
+  // It takes a list of rows like this:
+  //
+  //   +-------+-------+-------+--------+
+  //   | id    | other | name  | value  |
+  //   +-------+-------+-------+--------+
+  //   | one   | foo   | name1 | value1 |
+  //   | two   | bar   | name1 | value1 |
+  //   | two   | bar   | name2 | value2 |
+  //   | three | baz   | NULL  | NULL   |
+  //   +-------+-------+-------+--------+
+  //
+  // And aggregates the name/data pairs into a JSON object like this:
+  //
+  //  [
+  //    { id: 'one', other: 'foo', agg: { 'name1': 'value1' }},
+  //    { id: 'two', other: 'bar', agg: { 'name1': 'value1', 'name2': 'value1' }},
+  //    { id: 'three', other: 'baz', agg: {}},
+  //  ]
+  //
+  // Note that correct operation depends on the rows being ordered by id column,
+  // so they can be aggregated in a single pass.
+  //
+  // The column names can be customized, in order to work with several different
+  // queries.
+
+  function aggregateNameValuePairs(rows, idColumn, nameColumn, valueColumn, aggColumn) {
```

Comment:
Is this something that warrants its own module with some dedicated unit tests?

Reply:
(the unit tests is really the bit I'm asking about here tbh, regardless of the separate module part)